subsys/metrics/src/main/java/org/commonjava/indy/metrics/zabbix/sender/IndyZabbixSender.java

```diff
@@ -247,200 +230,96 @@ private IndyZabbixSender( boolean bCreateNotExistHostGroup, boolean bCreateNotEx
-        JSONObject group = new JSONObject();
-        group.put( "groupid", hostGroupCache.get( hostGroup ) );
-        groups.add( group );
-
-        // "interfaces": [
-        // {
-        // "type": 1,
-        // "main": 1,
-        // "useip": 1,
-        // "ip": "192.168.3.1",
-        // "dns": "",
-        // "port": "10050"
-        // }
-        // ],
-
-        JSONObject interface1 = new JSONObject();
-        interface1.put( "type", 1 );
-        interface1.put( "main", 1 );
-        interface1.put( "useip", 1 );
-        interface1.put( "ip", ip );
-        interface1.put( "dns", "" );
-        interface1.put( "port", "10051" );
-
-        Request request = RequestBuilder.newBuilder()
-                                        .method( "host.create" )
-                                        .paramEntry( "host", host )
-                                        .paramEntry( "groups", groups )
-                                        .paramEntry( "interfaces", new Object[] { interface1 } )
-                                        .build();
-        JSONObject response = zabbixApi.call( request );
-        return response.getJSONObject( "result" ).getJSONArray( "hostids" ).getString( 0 );
-    }
-
-    private JSONArray getHost( String host )
-    {
-        JSONObject filter = new JSONObject();
-        filter.put( "host", new String[] { host } );
-        Request getRequest = RequestBuilder.newBuilder().method( "host.get" ).paramEntry( "filter", filter ).build();
-        JSONObject getResponse = zabbixApi.call( getRequest );
-        return getResponse.getJSONArray( "result" );
+        return null;
     }
 
     private String itemCacheKey( String host, String item )
     {
         return host + ":" + item;
     }
 
-    void checkItem( String host, String item, int vauleType )
+    String checkItem( String host, String item, int valueType )
     {
+
         try
         {
             if ( itemCache.get( itemCacheKey( host, item ) ) == null )
             {
                 this.zabbixApiInit();
-                JSONArray result = getItem( host, item );
-                if ( result.isEmpty() )
+
+                String itemid = zabbixApi.getItem( host, item, hostCache.get( host ) );
+                if ( itemid == null )
                 {
-                    String itemId = createItem( host, item, vauleType );
-                    itemCache.put( itemCacheKey( host, item ), itemId );
+                    zabbixApi.createItem( host, item, hostCache.get( host ), valueType );
+                    itemCache.put( itemCacheKey( host, item ), itemid );
                 }
                 else
                 {
                     // put into cache
-                    itemCache.put( itemCacheKey( host, item ), item );
+                    itemCache.put( itemCacheKey( host, item ), itemid );
                 }
+                return itemid;
             }
         }
-        finally
+        catch ( Throwable throwable )
```

Comment:
Seems like in some places you catch IOException, and in others, Throwable. Is there some specific reason you need to catch literally everything (including InterruptedException and OutOfMemoryError) here?